src/components/molecules/EntranceStepsInputFieldSet/EntranceStepsInputFieldSet.tsx

```diff
@@ -30,6 +36,7 @@ export interface EntranceStepsInputFieldSetProps {
   onUpdate: UseArrayUpdate<EntranceStepConfig>;
   onRemove: UseArrayRemove<EntranceStepConfig>;
   register: (Ref: unknown, RegisterOptions?: unknown) => void;
+  setDirtyButtons: Dispatch<SetStateAction<boolean>>;
```

Comment:
Extra dependency that isn't needed. `onUpdate` and `onRemove` already exist and can be used to handle the dirty state in the parent. Also, they are made to return a context object which aside from `index` can be extended to return any other field of interest.

```suggestion
```